feat(platform): add support for the SCM-manager (#26866)
renovatebot/renovate: lib/modules/platform/scm-manager/readme.md

# SCM-Manager

Renovate supports the [SCM-Manager](https://scm-manager.org) platform.

## Authentication

> 1. Create an API Key for your technical Renovate user in SCM-Manager.
> 2. The technical user _must_ have valid name and email address.
> 3. Put the API key in the `RENOVATE_TOKEN` environment variable, so that Renovate can use it.

Reviewer: ```suggestion
1. Create an API Key for your technical Renovate user in SCM-Manager
1. The technical user _must_ have a valid name and email address
1. Put the API key in the `RENOVATE_TOKEN` environment variable, so that Renovate can use it
```

> You must set the `platform` config option to `scm-manager` in your Renovate config file.

Reviewer: I would like us to link to the docs. I know you removed the link because you could not figure out the correct format. Let's try again. 🙃 

@viceice can you suggest the correct link to the `platform` self-hosted config option docs?

https://docs.renovatebot.com/self-hosted-configuration/#platform

Reviewer 2: https://github.com/renovatebot/renovate/blob/f39568d8a809a3b170ecb812f21e38c1689e1ce0/lib/modules/platform/azure/readme.md#L128

```md
[`platform`](../../../configuration-options.md#platform)
```

> You must set the [`platform`](https://docs.renovatebot.com/self-hosted-configuration/#platform) to `scm-manager` in your Renovate config file.

Reviewer: ```suggestion
You must set the [`platform` config option](https://docs.renovatebot.com/self-hosted-configuration/#platform) to `scm-manager` in your Renovate config file.
```

> The technical user needs the permissions to read and write your repository.

Reviewer: ```suggestion
The technical user must have permission to read and write to your repository.
```

> This can be achieved by granting the permission role "OWNER" to your technical Renovate user.

Reviewer: ```suggestion
You can do this by granting the permission role "OWNER" to the technical Renovate user.
```

Additionally, the Review Plugin needs to be installed.
Otherwise, the pull request API will not be available.
Plugins can be installed under Administration -> Plugins -> Available.

> Renovate supports SCM-Manager major version 2.x and 3.x.
> The 2.x is supported since 2.48.0.
> The 3.x is supported since 3.0.0.

Reviewer: ```suggestion
Renovate supports SCM-Manager major version `2.x` and `3.x`.
The minimum version for the `2.x` range is `2.48.0`.
The minimum version for the `3.x` range is `3.0.0`.
```

> You must set the [`platform`](../../../self-hosted-configuration.md#platform) config option to `scm-manager` in your Renovate config file.
> 
> The technical user must have permission to read and write to your repository.
> You can do this by granting the permission role "OWNER" to the technical Renovate user.
> Additionally, the Review Plugin needs to be installed.
> Otherwise, the pull request API will not be available.
> Plugins can be installed under Administration -> Plugins -> Available.
> 
> Renovate supports SCM-Manager major version `2.x` and `3.x`.
> The minimum version for the `2.x` range is `2.48.0`.
> The minimum version for the `3.x` range is `3.0.0`.

Reviewer: ```suggestion
## Set correct platform

You must set the [`platform`](../../../self-hosted-configuration.md#platform) config option to `scm-manager` in your Renovate config file.

## Set permissions

The technical user must have permission to read and write to your repository.
You can do this by granting the permission role "OWNER" to the technical Renovate user.

## Install Review Plugin

To let Renovate access the Pull Request API, you must install the Review Plugin.
Find the list of available plugins by going to to Administration -> Plugins -> Available.

## Supported versions of SCM-Manager

Renovate supports SCM-Manager major version `2.x` and `3.x`.

The minimum version for the `2.x` range is `2.48.0`.
The minimum version for the `3.x` range is `3.0.0`.
```

**Changes:**

- Add headings
- Rewrite some text

**Questions:**

Is there just one "Review plugin" that the user must install? Or are there multiple apps that do the same thing?

If only one app will work, we should link to that app. If multiple apps work, create a list with the most used ones, and link to them.

# SCM-Manager

> Renovate supports the [SCM-Manager](https://scm-manager.org) platform.

Reviewer 2: ```suggestion
Renovate supports the [SCM-Manager](https://scm-manager.org) platform.
This platform is considered experimental
```

# SCM-Manager

Renovate supports the [SCM-Manager](https://scm-manager.org) platform.
This platform is considered experimental.

## Authentication

1. Create an API Key for your technical Renovate user in SCM-Manager
1. The technical user _must_ have a valid name and email address
1. Put the API key in the `RENOVATE_TOKEN` environment variable, so that Renovate can use it

## Set correct platform

You must set the [`platform`](../../../self-hosted-configuration.md#platform) config option to `scm-manager` in your Renovate config file.

## Set permissions

The technical user must have permission to read and write to your repository.

> You can do this by granting the permission role "OWNER" to the technical Renovate user.

Reviewer 3: I don't think we want to recomment this.
